chainstate/merkle_root.go

```diff
@@ -0,0 +1,63 @@
+package chainstate
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/libsv/bitcoin-hc/transports/http/endpoints/api/merkleroots"
+)
+
+// VerifyMerkleRoots will try to verify merkle roots with all available providers
+func (c *Client) VerifyMerkleRoots(ctx context.Context, merkleRoots []string) (*merkleroots.MerkleRootsConfirmationsResponse, error) {
+	merkleRootsRes, err := verifyMerkleRootsWithPulse(ctx, c, merkleRoots)
+
+	if err != nil {
+		debugLog(c, "", fmt.Sprintf("verify merkle root error: %s from Pulse", err))
+		return nil, err
+	}
+	debugLog(c, "", "successful verification of merkle proofs by Pulse")
+	return merkleRootsRes, nil
+}
+
+func verifyMerkleRootsWithPulse(ctx context.Context, c *Client, merkleProofs []string) (*merkleroots.MerkleRootsConfirmationsResponse, error) {
+	jsonData, err := json.Marshal(merkleProofs)
+
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	req, err := http.NewRequestWithContext(ctx, "POST", c.options.config.pulseClient.url, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Authorization", c.options.config.pulseClient.authToken)
```

Comment:
Hmmm... before setting this header, we should probably check if auth token isn't empty.
Because we always can deploy the Pulse without authorisation, so the token will be useless.

Beside that. We should prepend the token with `Bearer ` prefix so the "user" of the library doesn't need to provide the whole value of the header because it's rather misleading.

Reply:
Done